File: api/src/auth/auth.service.ts
import { Injectable } from '@nestjs/common';
import { LoginDto } from './dto/login.dto';
import { SignupDto } from './dto/signup.dto';

@Injectable()
export class AuthService {

  signup(dto: SignupDto) {
    return {
      message: 'signup success',
      user: dto
    };
  }

  login(dto: LoginDto) {
    return {
      message: 'login success'
    };
  }

}

File: server/src/main/java/com/bookmyvenue/server/auth/service/AuthService.java
package com.bookmyvenue.server.auth.service;

import com.bookmyvenue.server.auth.dto.request.LoginRequest;
import com.bookmyvenue.server.auth.dto.request.RegisterRequest;
import com.bookmyvenue.server.auth.dto.response.AuthResult;
import com.bookmyvenue.server.common.response.MessageResponse;


public interface AuthService {

    /**
     * Registers a new USER or VENDOR account.
     */
    MessageResponse register(RegisterRequest request);

    /**
     * Authenticates a user and returns JWT tokens.
     */
    AuthResult login(LoginRequest request);


    /**
     * Generates a new access token using a valid refresh token.
     */
    AuthResult refreshToken(String refreshToken);


    MessageResponse forgotPassword(String email);

    MessageResponse resetPassword(String email,String otp, String newPassword);
}